shared/src/vyxal/MiscHelpers.scala

```diff
@@ -199,4 +203,18 @@ object MiscHelpers:
       case (a: VList, b: VList) => VList.from((a, b).zipped.map(subtract(_, _)))
       case (a, b) => ???
 
+  def untilNoChange(function: VFun, value: VAny)(using Context): VList =
+    var prev = value
+    val res = LazyList.unfold(value) { curr =>
+      val next = function(curr)
+      if next == prev then None
+      else
+        prev = next
+        Some(next -> next)
+    }
+    VList.from(value #:: res)
+
+  def zipWith(left: VList, right: VList, function: VFun)(using Context): VList =
+    VList.from(left.zip(right).map((a, b) => function(a, b)))
```

Comment:
This should probably work:
```suggestion
    left.zipWith(right) { (a, b) => function(a, b) }
```
If we do end up adding `.asFn2` later, we could write `left.zipWith(right)(function.asFn2)`